1-Draft/RFCNNNN-Implicit-Line-Continuance.md

```diff
@@ -0,0 +1,148 @@
+
+For example, consider this script:
+
+```PowerShell
+function -dash {
+    'run'
+}
+
+Get-Process -Id $PID
+-dash
+```
+
+In PowerShell 6.x and earlier, two commands will be executed: `Get-Process`, and `-dash`. With this PR in place and the experimental feature enabled, only one command would be executed: `Get-Process`. The reason is that the parser would identify `-dash` as being intended as a parameter on the command on the previous line. That's how the parser needs to work to make this implicit line continuance functional.
+
+Similarly, consider this script:
+
+```PowerShell
+Get-Process -Id $PID
+-split 'a b c d'
+```
+
+Similar to the previous example, in PowerShell 6.x and earlier, one command and one statement will be executed: `Get-Process`, and the unary `-split` statement. With this PR in place and the experimental feature enabled, only one command would be executed: `Get-Process`, because again, the parser would identify `-split` as being intended as a parameter on the command on the previous line. The same applies to the `-join` unary operator as well. It does not apply to the `--` prefix arithmetic operator because the parser knows parameter names cannot start with a dash.
+
+To fix this, users can do one of the following:
+
+1. For either example, insert a blank line between `Get-Process` and the next command that starts with dash, as shown here:
+
+    ```PowerShell
+    function -dash {
+        'run'
+    }
+
+    Get-Process -Id $PID
+
+    -dash # Runs the -dash command because implicit line continuance stops looking when it encounters a blank line
+
+    Get-Process -Id $PID
+
+    -split 'a b c d' # splits the string 'a b c d' into an array with four items
+    ```
+
+1. In the example with a command that starts with a dash, invoke the command using the call operator, as shown here:
+
+    ```PowerShell
+    function -dash {
+        'run'
+    }
+
+    Get-Process -Id $PID
+    & -dash # Runs the `-dash` command because implicit line continuance sees the call operator and recognizes that the line does not continue further
+    ```
+
+    Unfortunately this workaround does not work for the -split or -join unary operators, because you cannot invoke statement that starts with a unary operator with the call operator (the call operator is only for invoking commands).
+
+In practice, I believe that there are not many commands out there that start with a dash. While researching this, I couldn't find a single command on Windows or Linux that starts with a dash, so that's not a very risky scenario. However, since the `-split` and `-join` unary operators exist in PowerShell, there is a good likelihood that those may be used on a line following a command, and that is where this breaking change would have the most potential impact. While I don't feel that risk is enough to reject this proposal entirely, because it adds significant value to script authors, it is worth considering what the best approach would be.
```

Comment:
I can't imagine there are **many** commands using operators as parameter names, but I imagine there may be some where "Join" is the first word in the parameter...

Reply:
Agreed. Better to avoid special casing anything like that and come up with a way to use such functionality without breaking existing scripts/modules.